### ISSUE TITLE
debug watchers should avoid duplicate display names

### ISSUE THREAD

**Issue by Geokureli** [HaxeFlixel/flixel#2896]
The following adds multiple watchers but the second should replace the first
```hx
FlxG.watch.addFunction("foo", ()->5);
FlxG.watch.addFunction("foo", ()->10);
```

**Comment by MemeHovy** [comment]
wouldn't it get overwritten though? by looking at [Watch.hx#L47](https://github.com/HaxeFlixel/flixel/blob/dev/flixel/system/debug/watch/Watch.hx#L47)

**Comment by Geokureli** [comment]
> wouldn't it get overwritten though? judging from by looking at [Watch.hx#L47](https://github.com/HaxeFlixel/flixel/blob/dev/flixel/system/debug/watch/Watch.hx#L47)

only for `addQuick`, not `addFunction`

**Comment by ninjamuffin99** [comment]
fixed via #2902
